internal/file/file_manager_service.go

```diff
@@ -73,7 +111,34 @@ func (fms *FileManagerService) UpdateFile(
 		},
 	}
 
-	_, updateFileErr := fms.fileServiceClient.UpdateFile(ctx, request)
+	backOffCtx, backoffCancel := context.WithTimeout(ctx, fms.agentConfig.Common.MaxElapsedTime)
+	defer backoffCancel()
 
-	return updateFileErr
+	sendUpdateFile := func() (*mpi.UpdateFileResponse, error) {
```

Comment:
same comment here, for readability, maybe have it as it's own function at the file level

Reply:
same answer as above